internal/controller/nim/account_controller.go

```diff
@@ -383,6 +449,52 @@ func (r *AccountReconciler) reconcileNimPullSecret(
 	return secret, nil
 }
 
+// getSelectedModelList returns a list of Ids of the selected models
+func (r *AccountReconciler) getSelectedModelList(
+	ctx context.Context, cmRef *corev1.ObjectReference,
+	namespace string) ([]string, error) {
+	logger := log.FromContext(ctx)
+	logger.V(1).Info("getting selected model list")
+
+	// if selected model list is not set
+	if cmRef == nil || len(cmRef.Name) == 0 {
+		return []string{}, nil
+	}
+
+	// get the config map that contains the selected model list
+	cmNs := cmRef.Namespace
+	if cmNs == "" {
+		cmNs = namespace
+	}
+	modelListCm := &corev1.ConfigMap{}
+	modelListCmSubject := types.NamespacedName{Name: cmRef.Name, Namespace: cmNs}
+	if err := r.Client.Get(ctx, modelListCmSubject, modelListCm); err != nil {
+		if k8serrors.IsNotFound(err) {
+			logger.Error(err, "failed to fetch the config map for getting the selected model list",
+				"config map", modelListCmSubject)
+			return []string{}, nil
+		}
+		return nil, err
+	}
+
+	// convert the selected model list to a string array
+	if modelListCm.Data != nil {
+		if models, ok := modelListCm.Data["models"]; ok {
+			var selectedModelIds []string
+			if err := json.Unmarshal([]byte(models), &selectedModelIds); err != nil {
+				logger.Error(err, "failed to unmarshal the selected mode list",
+					"model list", models)
+				return []string{}, nil
```

Comment:
shouldn't the error be returned here?

Reply:
Returning an error from here, will re-queue the reconcilation request, and we'll be stuck here in a loop until the user fixes the issue with the ConfigMap. But if we log the error and return `[]string{}, nil` the controller will procceed as if the custom user ConfigMap doesn't exist.

Reply:
See comment: https://github.com/opendatahub-io/odh-model-controller/pull/405#discussion_r2024896907

Reply:
ok